misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -4181,22 +4223,40 @@ private List<Minutiae> removeTrailingWS(List<Minutiae> trailingMinutiae) {
     }
 
     /**
-     * Indent the code by the 4-whitespace characters.
+     * Indent the code by the number of white-spaces defined by tab-size.
      */
     private void indent() {
-        env.currentIndentation += options.getTabSize();
+        indent(1);
+    }
+
+    /**
+     * Indent the code by the number of white-spaces defined by tab-size.
+     *
+     * @param step Number of tabs.
+     */
+    private void indent(int step) {
+        env.currentIndentation += (options.getTabSize() * step);
     }
 
     /**
-     * Undo the indentation of the code by the 4-whitespace characters.
+     * Undo the indentation of the code by the number of white-spaces defined by tab-size.
+     */
+    private void unIndent() {
+        unIndent(1);
+    }
+
+    /**
+     * Undo the indentation of the code by the number of white-spaces defined by tab-size.
+     *
+     * @param step Number of tabs.
      */
-    private void unindent() {
-        if (env.currentIndentation < options.getTabSize()) {
+    private void unIndent(int step) {
```

Comment:
Shall we leave the method name as `unindent` since it is one word?

Reply:
will do that